### PR TITLE
`P256_VERIFY` edge cases testing 

### LINKED ISSUE #2516: `P256_VERIFY` edge cases testing
Tests related to gas are already available in:
- `SixtyThreeSixtyFourthsPrecompileTests`;
- `P256VerifyCallSuccessAndFailureCasesTest`;
- `LowGasStipendPrecompileCallTests`;

Test related to size are already available in:

- `PostCancunPrecompileSizeTest`;

Test vector from the EIP:
- `P256VerifyTest`;

### PR DESCRIPTION
**Pull request by lorenzogentile404**
> [!NOTE]
> Adds comprehensive P256_VERIFY precompile tests (edge cases, gas scenarios) and a helper to capture RETURNDATASIZE, refactoring call tests to use constants and validate hub behavior post-Osaka.
> 
> - **Tests — P256_VERIFY**:
>   - Add extensive edge-case suites exploring `r`, `s`, `qX`, `qY` boundaries and verify ECData return data post-Osaka.
>   - Introduce scenarios: insufficient gas, stipend-enabled success, invalid/valid call data sizes, internal-checks fail/succeed but signature fails, and fully valid verification.
>   - Use `EIP_7825_TRANSACTION_GAS_LIMIT_CAP` for calls; assert `P256VerifyOobCall` hub success/failure and returned data size/content.
> - **Refactor**:
>   - Replace direct calls with `simpleCallAndReturnDataSize(...)` and constants (`GAS_CONST_P256_VERIFY`, `GAS_CONST_G_CALL_STIPEND`, `PRECOMPILE_*`).
>   - `testP256VerifyBody` now returns `BytecodeRunner` and accepts optional trailing bytecode for stack/return-data checks.
> - **Utilities**:
>   - Add `Utilities.simpleCallAndReturnDataSize(...)` to perform a call then `RETURNDATASIZE`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 4e1e394fd8989947c73727db9f35e622b705aa39. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/P256VerifyTest.java [hunk 1]
```diff
@@ -128,4 +148,120 @@ private static Stream<Arguments> p256VerifySourceNightly() throws IOException {
...
+          }
+        }
+      }
+    }
+    return arguments.stream();
+  }
+
+  // Edge cases with checks over return data
+  @Test
+  void testIcpIsFalseP256Verify(TestInfo testInfo) {
```
Comment (OlivierBBB) [inline arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/P256VerifyTest.java]: `testInternalChecksFailP256Verify`

File: arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/P256VerifyTest.java [hunk 2]
```diff
@@ -128,4 +148,120 @@ private static Stream<Arguments> p256VerifySourceNightly() throws IOException {
...
+      P256VerifyOobCall p256VerifyOobCall =
+          (P256VerifyOobCall)
+              bytecodeRunner.getHub().oob().operations().stream().toList().getLast().oobCall();
+      assertTrue(p256VerifyOobCall.isHubSuccess());
+
+      EcDataOperation ecDataOperation = bytecodeRunner.getHub().ecData().ecDataOperation();
+      assertFalse(ecDataOperation.internalChecksPassed());
+
+      final Bytes returnDataSize = bytecodeRunner.getHub().currentFrame().frame().getStackItem(0);
+      assertTrue(returnDataSize.isZero());
```
Comment (OlivierBBB) [inline arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/P256VerifyTest.java:211]: What about asserting the `callSuccess` of the `HUB` ?

File: arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/P256VerifyTest.java [hunk 3]
```diff
@@ -128,4 +148,120 @@ private static Stream<Arguments> p256VerifySourceNightly() throws IOException {
...
+    }
+  }
+
+  @Test
+  void testIcpIsTrueButSignatureVerificationFailsP256Verify(TestInfo testInfo) {
```
Comment (OlivierBBB) [inline arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/P256VerifyTest.java]: `testInternalChecksPassButSignatureVerificationFailsP256Verify`

File: arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/P256VerifyTest.java [hunk 4]
```diff
@@ -128,4 +148,120 @@ private static Stream<Arguments> p256VerifySourceNightly() throws IOException {
...
+      P256VerifyOobCall p256VerifyOobCall =
+          (P256VerifyOobCall)
+              bytecodeRunner.getHub().oob().operations().stream().toList().getLast().oobCall();
+      assertTrue(p256VerifyOobCall.isHubSuccess());
+
+      EcDataOperation ecDataOperation = bytecodeRunner.getHub().ecData().ecDataOperation();
+      assertTrue(ecDataOperation.internalChecksPassed());
+
+      final Bytes returnDataSize = bytecodeRunner.getHub().currentFrame().frame().getStackItem(0);
+      assertTrue(returnDataSize.isZero());
```
Comment (OlivierBBB) [inline arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/P256VerifyTest.java:241]: Same checks here

File: arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/P256VerifyTest.java [hunk 5]
```diff
@@ -128,4 +148,120 @@ private static Stream<Arguments> p256VerifySourceNightly() throws IOException {
...
+      P256VerifyOobCall p256VerifyOobCall =
+          (P256VerifyOobCall)
+              bytecodeRunner.getHub().oob().operations().stream().toList().get(1).oobCall();
+      assertTrue(p256VerifyOobCall.isHubSuccess());
+
+      EcDataOperation ecDataOperation = bytecodeRunner.getHub().ecData().ecDataOperation();
+      assertTrue(ecDataOperation.internalChecksPassed());
+
+      final Bytes returnDataSize = bytecodeRunner.getHub().currentFrame().frame().getStackItem(0);
+      assertFalse(returnDataSize.isZero());
```
Comment (OlivierBBB) [inline arithmetization/src/test/java/net/consensys/linea/zktracer/module/ecdata/P256VerifyTest.java:276]: You could assert it's `32`

**Review by OlivierBBB — APPROVED** [review]
LGTM.

Should we append 32 `JUMPDEST`'s after the `CALL` / after the `RETURNDATASIZE` ?
